### ISSUE TITLE
Onboarding process -- theme/plugin install/activate without capability? No way to remove?

### ISSUE THREAD

**Issue by tripflex** [woocommerce/woocommerce#32272]
A have a multisite installation where users are NOT allowed to install plugins or themes, but are able to activate and use WooCommerce (and plugins already installed).

Is there not any way that I can remove the parts of the onboarding that promotes paid (and free) plugins to be installed, as well as the theme step?

I don't see any filters or anywhere to modify the steps, it looks like they are all hard coded.  The user also does NOT have `install_themes` or `install_plugins` capability, but it still shows on the onboarding.

I was able to remove the paid addon promotion using `woocommerce_admin_onboarding_product_types` to unset them.

I was able to remove all themes except current one from showing using `woocommerce_admin_onboarding_themes` -- even though the user does not have `switch_themes` capability.

It also still shows the "Upload a Theme" even though user does not have `install_themes` capability

I would like to still be able to offer the onboarding functionality to users, but it seems this was built with the intention of promoting addon plugins (free and paid) without checking the user's capabilities, and I don't see any way to decouple that without forking and building my own version.

**_Please tell me i'm wrong and just not seeing something?_**

Specifically this appears to be hard coded, and even though those plugins were removed via the `woocommerce_admin_onboarding_plugins_whitelist` and `woocommerce_admin_plugins_whitelist` it still shows on the onboarding.
https://github.com/woocommerce/woocommerce-admin/blob/70f45daca16183e0b7d29e69981bf2250eb8d31c/client/profile-wizard/steps/business-details.js#L385-L416

As well as the `fashion-apparel-accessories` or `health-beauty` industries being selected to promote the installation of the plugins as well (I was able to get past that by removing those industries using a filter, but not ideal).

**Comment by joshuatf** [comment]
I think this is somewhat important and while it's not necessarily a security risk, it exposes UI that should not be exposed to certain users.  Going to move this into our cycle backlog.

**Comment by chihsuan** [comment]
> I would like to still be able to offer the onboarding functionality to users, but it seems this was built with the intention of promoting addon plugins (free and paid) without checking the user's capabilities, and I don't see any way to decouple that without forking and building my own version.

I think this is something we should keep in mind and we can add a check in our new OBW. cc @moon0326

**Comment by moon0326** [comment]
Thanks for the ping @chihsuan 

We should definitely provide extensibility in the new OBW.

**Comment by adrianduffell** [comment]
I think we could just skip the Free Features screen when the user does not have the capability to install plugins.

This would solve the managed multisite use case in the OP where users are restricted from add plugins. I don't think it's worth showing a "contact your administrator" type message because it's not critical that the plugins are installed. Similarly we wouldn't want the plugin list to be filterable by hosts because the recommendations should come from Woo.

**Comment by moon0326** [comment]
Hey @rjchow ✋ ,

I was going to work on it, but I realized you're working on refactor. Should I wait until your refactor is done?

**Comment by rjchow** [comment]
Yes please! Also I think there’s a bit to unpack in the original issue - sounds like we might want to consider expanding it into a few tasks to work on, that might tie in with our upcoming focuses.

- capability check guard
- extensibility via plugin add/remove filters 
- extensibility via pages add/remove

@adrianduffell what do you think?

**Comment by moon0326** [comment]
@rjchow 

Sounds good!

> capability check guard

I think this can be done from the same API endpoint. We could return a status code related to permissions from the extension list API.

**Comment by psealock** [comment]
@rjchow I believe this issue is closed by https://github.com/woocommerce/woocommerce/pull/52301, is that right?

**Comment by rjchow** [comment]
@psealock I didn't have this issue in mind when working on that PR, but yes it would seem to at least partially address it by showing an error screen preventing any installation.

The OP in this issue is asking for an option for the screen not to be shown at all, however. (Also I believe they are referring to the old OBW).

**Comment by tripflex** [comment]
@psealock @rjchow my biggest concern here is that there was no way to remove the promotion of plugins to be installed (paid and free), there should be at least some way of doing this

**Comment by chihsuan** [comment]
> my biggest concern here is that there was no way to remove the promotion of plugins to be installed (paid and free), there should be at least some way of doing this

@tripflex In the latest version, only free plugins are displayed on the onboarding screen; paid plugins are not included so i

Regarding the ability to remove the promotion of plugins to be installed, @adrianduffell commented:

> Similarly we wouldn't want the plugin list to be filterable by hosts because the recommendations should come from Woo.

Let us know if you have any other suggestions or feedback.
